pkg/logs/input/docker/tailer.go

```diff
@@ -185,19 +205,25 @@ func (t *Tailer) readForever() {
 					// This error is raised when the agent is stopping
 					return
 				case err == io.EOF:
```

Comment:
Here I would have reused the same logic as in `case isContextCanceled(err):` and reused the method `wait()` to avoid hammering the docker socket and avoid the CPU to go high.

Reply:
should we only try once to setup the reader than ask the launcher to restart the tailer?

Reply:
in general, does it make sense to have different retry strategies depending on the error (`isContextCanceled`, `io.EOF`, and `default`)? If so, why? (this is an honest question, I don't know these different types of errors enough to answer).

Otherwise I feel all errors that require a retry could use the same overall logic: retry setting up the reader, and if that failed ask the launcher to restart the tailer.

Reply:
I have the same feeling.
pr updated in this direction.

Reply:
agreed